utils/index.ts

```diff
@@ -215,3 +218,29 @@ export function getImageResizeURL (url: string, { width = 300 }: { width?: numbe
   const { LIKE_CO_STATIC_ENDPOINT } = useRuntimeConfig().public
   return `${LIKE_CO_STATIC_ENDPOINT}/thumbnail/?url=${encodeURIComponent(url)}&width=${width}`
 }
+
+export function fileToArrayBuffer (file: Blob): Promise<string | ArrayBuffer | null> {
+  return new Promise((resolve) => {
+    const reader = new FileReader()
+    reader.onload = () => resolve(reader.result)
+    reader.readAsArrayBuffer(file)
+  })
+}
+
+export async function digestFileSHA256 (buffer: ArrayBuffer) {
+  const hashBuffer = await crypto.subtle.digest('SHA-256', buffer)
+  const hashArray = Array.from(new Uint8Array(hashBuffer))
+  const hashHex = hashArray.map(b => b.toString(16).padStart(2, '0')).join('')
+  return hashHex
```

Comment:
Is this just converting buffer to hex? can we just use `Buffer.from().toString('hex')`?

Reply:
[🐛 Refactor digestFileSHA256 to use Buffer for hash calculation](https://github.com/likecoin/nft-book-press/pull/290/commits/e94d7a74ccaa50d9680f6188ea3e41c8ea8907e3)